File: src/main/scala/com/szadowsz/gospel/core/data/socket/AbstractSocket.scala
package com.szadowsz.gospel.core.data.socket

import java.net.InetAddress
import java.{util => ju}

import com.szadowsz.gospel.core.data.{Term, Var}

@SerialVersionUID(1L)
abstract class AbstractSocket extends Term {
  def isClientSocket: Boolean

  def isServerSocket: Boolean

  def isDatagramSocket: Boolean

  def getSocket: AnyRef

  def getAddress: InetAddress

  override def isEmptyList: Boolean = false

  override def isAtomic: Boolean = false

  override def isCompound: Boolean = false

  override def isAtom: Boolean = false

  override def isList: Boolean = false


  override def isGround: Boolean = false

  override def isGreater(t: Term): Boolean = false


  override def isGreaterRelink(t: Term, vorder: ju.ArrayList[String]): Boolean = false

  override def isEqual(t: Term): Boolean = false

  override def getTerm: Term = this

  override def free {}

  private[gospel] override def resolveTerm(count: Long): Long = count


  /**
   * gets a copy (with renamed variables) of the term.
   * <p>
   * the list argument passed contains the list of variables to be renamed
   * (if empty list then no renaming)
   */
  private[gospel] override def copy(vMap: ju.AbstractMap[Var, Var], idExecCtx: Int): Term = this


  /**
   * gets a copy of the term.
   */
  private[gospel] override def copy(vMap: ju.AbstractMap[Var, Var], substMap: ju.AbstractMap[Term, Var]): Term = this
}

File: tuProlog 4.0.0 - SNAPSHOT/src/alice/tuprolog/AbstractSocket.java
package alice.tuprolog;
import java.net.InetAddress;
import java.util.AbstractMap;
//import java.util.ArrayList;

import alice.tuprolog.interfaces.TermVisitor;

public abstract class AbstractSocket extends Term{
	
	private static final long serialVersionUID = 1L;
	
	public abstract boolean isClientSocket();
	
	public abstract boolean isServerSocket();
	
	public abstract boolean isDatagramSocket();
	
	public abstract Object getSocket();
	
	public abstract InetAddress getAddress();
	
	@Override
	public boolean isNumber() {
		return false;
	}

	@Override
	public boolean isStruct() {
		return false;
	}

	@Override
	public boolean isVar() {
		return false;
	}

	@Override
	public boolean isEmptyList() {
		return false;
	}

	@Override
	public boolean isAtomic() {
		return false;
	}

	@Override
	public boolean isCompound() {
		return false;
	}

	@Override
	public boolean isAtom() {
		return false;
	}

	@Override
	public boolean isList() {
		return false;
	}

	@Override
	public boolean isGround() {
		return false;
	}

	@Override
	public boolean isGreater(Term t) {
		return false;
	}

	@Override
	public boolean isEqual(Term t) {
		return false;
	}

	@Override
	public Term getTerm() {
		return this;
	}

	@Override
	public void free() {
	}

	@Override
	long resolveTerm(long count) {
		return count;
	}

	@Override
	Term copy(AbstractMap<Var, Var> vMap, int idExecCtx) {
		return this;
	}

	@Override
	Term copy(AbstractMap<Var, Var> vMap, AbstractMap<Term, Var> substMap) {
		return this;
	}
	
	@Override //Alberto
	public Term copyAndRetainFreeVar(AbstractMap<Var,Var> vMap, int idExecCtx) {
		return this;
	}

	@Override
	public void accept(TermVisitor tv) {
	}
}